jsonalchemy/json_encoder.py

```diff
@@ -0,0 +1,12 @@
+import json
+
+from jsonalchemy.dictstyle import JSONDict, JSONList
+
+class JSONAlchemyEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, JSONDict):
+            return {key: self.default(value) for key, value in dict(obj).iteritems()}
+        elif isinstance(obj, JSONList):
+            return [self.default(el) for el in obj]
+        else:
+            return obj
```

Comment:
The idea here is that we expect to get a `JSONDict` or a `JSONList`, which contains a combination of those types plus `str`, `int`, and a few other basic types. If we get one of the former, then we have to instruct `JSONEncoder` how to parse it. If we get one of the latter, then `JSONEncoder` already knows how to do it (for example, `None` becomes `null`).

Maybe I should add a clarifying comment?